forgerock-openbanking-uk-aspsp-common/src/main/java/com/forgerock/openbanking/common/model/data/OBEventNotification2.java

```diff
@@ -0,0 +1,290 @@
+/**
+ * Copyright 2019 ForgeRock AS.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.common.model.data;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import org.springframework.validation.annotation.Validated;
+import uk.org.openbanking.datamodel.event.OBEvent1;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.Objects;
+
+/**
+ * The resource event
+ * Only to import events, we have change the class name generated to avoid conflicts with the openbanking spec model
```

Comment:
Why are we doing this? Does it have fields that aren't in the generated model object? Or is it to enable us to accommodate future OBEventNotification objects in the spec?

What is the original generated class called? Should we prefix this one with `FR` to help distinguish it?

Reply:
In EventNotificationAPISpecification v3.1.2:
https://openbanking.atlassian.net/wiki/spaces/DZ/pages/1077806617/Event+Notification+API+Specification+-+v3.1.2#EventNotificationAPISpecification-v3.1.2-OBEventNotification2
They describe the object `OBEventNotification2` to used such as notification event class that contains the events claim but this object isn't defined in any swagger json file descriptor and there is only the object OBEventNotification1.

For that reason I created this OBEventNotification2 object only with the finality to import the notifications by `/api/data/events` and is a collection on FRDataEvents object created for this feature, only to use it for import, export, update and delete events by /api/data/events.

In Short the original class internally is OBEventNotification1 created from swagger file and is used around all source related with the events except for this new feature that is a specific feature for the customer and the objects used in are isolated only for that feature in a dedicated model/data package.

It's weird I know but doesn't exist this OBEventNotification2 in any swagger file:
- https://openbanking.atlassian.net/wiki/spaces/DZ/pages/1077806750/Real+Time+Event+Notification+API+-+v3.1.2
- https://raw.githubusercontent.com/OpenBankingUK/read-write-api-specs/v3.1.2-RC1/dist/event-notifications-swagger.json

If in a future we need to use this OBEventNotification2 we can move it to a new package to use it around the sandbox source but for the moment it's better still it isolated.

Reply:
Thanks for the useful explanation. I think it would have been good to place this in the javadoc, but I see you've merged it now.